src/dndbeyond/base/utils.js

```diff
@@ -174,39 +174,36 @@ function buildAttackRoll(character, attack_source, name, description, properties
             if (brutal > 0) {
                 const rule = parseInt(character.getGlobalSetting("critical-homebrew", CriticalRules.PHB));
                 let highest_dice = 0;
-                let homebrew_max_damage = 0;
-                if (rule == CriticalRules.HOMEBREW_MAX) {
-                    let highest_damage = 0;
-                    for (let dmg of crit_damages) {
-                        if (dmg > highest_damage){
-                            highest_damage = dmg;
-                        }
-                    }
-                    homebrew_max_damage = brutal * highest_damage;
-                } else {
-                    for (let dmg of crit_damages) {
+                let weapon_damage_counter = 0;
+                for (let dmg of damages) {
+                    if(weapon_damage_counter < weapon_damage_length){
```

Comment:
It's better to do `if (weapon_damage_counter >= weapon_damage_length) break;`
Also missing a space after the `if` (coding style)
also, I would check for weapon_damage_length since you init it at 0, so it should be `if (weapon_damage_length && weapon_damage_counter >= weapon_damage_length) break;`